pkg/scheduler/framework/framework_test.go

```diff
@@ -1393,3 +1448,205 @@ func TestCrossReferencePickedCustersAndObsoleteBindings(t *testing.T) {
+		ObjectMeta: metav1.ObjectMeta{
+			Name: altBindingName,
+			// Set the resource version; this is needed so that the calculated patch will not
+			// include the resource version field.
+			ResourceVersion: resourceVersion,
+		},
+		Spec: fleetv1beta1.ResourceBindingSpec{
+			TargetCluster:                clusterName,
+			SchedulingPolicySnapshotName: policyName,
+			ClusterDecision: fleetv1beta1.ClusterDecision{
+				ClusterName: clusterName,
+				Selected:    true,
+				ClusterScore: &fleetv1beta1.ClusterScore{
+					TopologySpreadScore: &topologySpreadScore,
+					AffinityScore:       &affinityScore,
+				},
+				Reason: pickedByPolicyReason,
+			},
+		},
+	}
+
+	policy := &fleetv1beta1.ClusterSchedulingPolicySnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: policyName,
+		},
+	}
+
+	fakeClient := fake.NewClientBuilder().
+		WithScheme(scheme.Scheme).
+		WithObjects(toPatchBinding, toDeleteBinding).
+		Build()
+	// Construct framework manually instead of using NewFramework() to avoid mocking the controller manager.
+	f := &framework{
+		client: fakeClient,
+	}
+
+	ctx := context.Background()
+
+	toCreate := []*fleetv1beta1.ClusterResourceBinding{toCreateBinding}
+	toPatch := []*bindingWithPatch{
+		{
+			updated: updatedBinding,
+			patch:   client.MergeFrom(toPatchBinding),
+		},
+	}
+	toDelete := []*fleetv1beta1.ClusterResourceBinding{toDeleteBinding}
+	if err := f.manipulateBindings(ctx, policy, toCreate, toDelete, toPatch); err != nil {
+		t.Fatalf("manipulateBindings() = %v, want no error", err)
+	}
+
+	// Check if the requested binding has been created.
+	createdBinding := &fleetv1beta1.ClusterResourceBinding{}
+	if err := fakeClient.Get(ctx, types.NamespacedName{Name: bindingName}, createdBinding); err != nil {
+		t.Errorf("Get() binding %s = %v, want no error", bindingName, err)
+	}
+	if diff := cmp.Diff(createdBinding, toCreateBinding, ignoreTypeMetaAPIVersionKindFields); diff != "" {
+		t.Errorf("created binding %s diff (-got, +want): %s", bindingName, diff)
+	}
+
+	// Check if the requested binding has been patched.
+	patchedBinding := &fleetv1beta1.ClusterResourceBinding{}
+	if err := fakeClient.Get(ctx, types.NamespacedName{Name: altBindingName}, patchedBinding); err != nil {
+		t.Errorf("Get() binding %s = %v, want no error", altBindingName, err)
+	}
+	if diff := cmp.Diff(patchedBinding, updatedBinding, ignoreTypeMetaAPIVersionKindFields); diff != "" {
+		t.Errorf("patched binding %s diff (-got, +want): %s", altBindingName, diff)
+	}
+
+	// Check if the requested binding has been deleted.
+	deletedBinding := &fleetv1beta1.ClusterResourceBinding{}
+	if err := fakeClient.Get(ctx, types.NamespacedName{Name: anotherBindingName}, deletedBinding); err != nil {
+		t.Errorf("Get() binding %s = %v, want no error", anotherBindingName, err)
+	}
+	if deletedBinding.Spec.State != fleetv1beta1.BindingStateUnscheduled {
+		t.Errorf("binding %s state = %s, want unscheduled", anotherBindingName, deletedBinding.Spec.State)
+	}
```

Comment:
compare the full structure to test we only update the state and other fields should not change?

Reply:
Done.